File: app/src/main/java/sk/styk/martin/apkanalyzer/util/BigDecimalFormatter.java
package sk.styk.martin.apkanalyzer.util;

import java.text.DecimalFormat;

/**
 * Created by Martin Styk on 31.07.2017.
 */
public class BigDecimalFormatter {
    private static DecimalFormat commonFormat;

    public static DecimalFormat getCommonFormat() {
        if (commonFormat == null) {
            commonFormat = new DecimalFormat();
            commonFormat.setMaximumFractionDigits(2);
            commonFormat.setMinimumFractionDigits(2);
        }
        return commonFormat;
    }

    public static DecimalFormat getFormat(int minFractions, int maxFractions) {
        DecimalFormat customFormat = new DecimalFormat();
        customFormat.setMaximumFractionDigits(maxFractions);
        customFormat.setMinimumFractionDigits(minFractions);

        return customFormat;
    }
}

File: app/src/main/java/sk/styk/martin/apkanalyzer/util/BigDecimalFormatter.kt
package sk.styk.martin.apkanalyzer.util

import java.text.DecimalFormat

object BigDecimalFormatter {
    private var commonFormat: DecimalFormat? = null

    fun getCommonFormat(): DecimalFormat {
        if (commonFormat == null) {
            commonFormat = getFormat(2, 2)
        }
        return commonFormat!!
    }

    fun getFormat(minFractions: Int, maxFractions: Int): DecimalFormat {
        val customFormat = DecimalFormat()
        customFormat.maximumFractionDigits = maxFractions
        customFormat.minimumFractionDigits = minFractions

        return customFormat
    }
}
